svgdigitizer/svgplot.py

```diff
@@ -204,22 +215,172 @@ class SVGPlot:
+            ...   </g>
+            ... </svg>'''))
+            >>> plot = SVGPlot(svg)
+            >>> plot.xlabel
+            'E'
+
+        """
+        [axis] = [
+            axis
+            for axis, orientation in self.axis_orientations.items()
+            if orientation == AxisOrientation.HORIZONTAL
+        ]
+        return axis
+
+    @property
+    @cache
+    def ylabel(self):
+        r"""
+        Return the label of the y axis.
+
+        EXAMPLES::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">E1: 0 cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 A</text>
+            ...   </g>
+            ... </svg>'''))
+            >>> plot = SVGPlot(svg)
+            >>> plot.ylabel
+            'j'
+
+        """
+        [axis] = [
+            axis
+            for axis, orientation in self.axis_orientations.items()
+            if orientation == AxisOrientation.VERTICAL
+        ]
+        return axis
+
+    @property
+    @cache
+    def axis_orientations(self):
+        r"""
+        Return the orientation for each axis.
+        The current implementation does not work for extrem cases where a plot is tilted by 45°.
+
+        EXAMPLES::
+
+            >>> from svgdigitizer.svg import SVG
+            >>> from io import StringIO
+            >>> svg = SVG(StringIO(r'''
+            ... <svg>
+            ...   <g>
+            ...     <path d="M 0 200 L 0 100" />
+            ...     <text x="0" y="200">E1: 0 cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M 100 200 L 100 100" />
+            ...     <text x="100" y="200">E2: 1cm</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 100 L 0 100" />
+            ...     <text x="-100" y="100">j1: 0</text>
+            ...   </g>
+            ...   <g>
+            ...     <path d="M -100 0 L 0 0" />
+            ...     <text x="-100" y="0">j2: 1 A</text>
+            ...   </g>
+            ... </svg>'''))
+            >>> plot = SVGPlot(svg)
+            >>> plot.axis_orientations
+            {'E': <AxisOrientation.HORIZONTAL: 'horizontal'>, 'j': <AxisOrientation.VERTICAL: 'vertical'>}
+
+        """
+        from numpy import trace
+        from numpy.linalg import det, qr
+
+        axis_orientations = {}
+        case1 = self._transformation(
```

Comment:
Since you are doing the same thing twice here, it should be possible to simplify this. Let me try.